DataDog/datadog-agent
# change logs_config.docker_container_use_file default to true

### What does this PR do?

Change `logs_config.docker_container_use_file` default to `true`. This will improve the performance and reliability of agents collecting logs from docker when the files are available and accessibly by the agent. (Followup from: https://github.com/DataDog/datadog-agent/pull/7109)

This PR also cleans up the status page when using `container_collect_all` so that the parent source is hidden when there are child (file) sources. 

### Describe how to test your changes

QA:

1. a host agent running as dd-agent with docker running. 
Should see a log line like: `Error accessing /var/lib/docker/containers, no such file or directory, falling back on tailing from Docker socket`
Check agent status that files are not being tailed. 

2. a container agent with the file paths passed in [see config here](https://docs.datadoghq.com/agent/docker/log/?tab=containerinstallation#installation)
Should not see the error. Check agent status that files are being tailed. 

### Checklist


- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The appropriate `team/..` label has been applied, if known.

Note: Adding GitHub labels is only possible for contributors with write access.
